### PR TITLE
Fix NVMe partition names

### PR DESCRIPTION
**Pull request by yh117**
When partitions are created on NVMe devices, the current udev rules are using partition IDs as the device IDs to assign `ID_SERIAL_SHORT`. For example, if the first partition is created on NVMe device `nvme0n5`, the `ID_SERIAL_SHORT` of the partition will be `local-nvme-ssd-0` intead of `local-nvme-ssd-4`. This PR will fix this issue.

### CODE REVIEW

File: src/lib/udev/rules.d/65-gce-disk-naming.rules [hunk 1]
```diff
@@ -21,7 +21,30 @@ SUBSYSTEM!="block", GOTO="gce_disk_naming_end"
...
+KERNEL=="nvme0n15*", ATTRS{model}=="nvme_card", ENV{ID_SERIAL_SHORT}="local-nvme-ssd-14"
+KERNEL=="nvme0n16*", ATTRS{model}=="nvme_card", ENV{ID_SERIAL_SHORT}="local-nvme-ssd-15"
+KERNEL=="nvme0n17*", ATTRS{model}=="nvme_card", ENV{ID_SERIAL_SHORT}="local-nvme-ssd-16"
+KERNEL=="nvme0n18*", ATTRS{model}=="nvme_card", ENV{ID_SERIAL_SHORT}="local-nvme-ssd-17"
+KERNEL=="nvme0n19*", ATTRS{model}=="nvme_card", ENV{ID_SERIAL_SHORT}="local-nvme-ssd-18"
+KERNEL=="nvme0n20*", ATTRS{model}=="nvme_card", ENV{ID_SERIAL_SHORT}="local-nvme-ssd-19"
+KERNEL=="nvme0n21*", ATTRS{model}=="nvme_card", ENV{ID_SERIAL_SHORT}="local-nvme-ssd-20"
+KERNEL=="nvme0n22*", ATTRS{model}=="nvme_card", ENV{ID_SERIAL_SHORT}="local-nvme-ssd-21"
+KERNEL=="nvme0n23*", ATTRS{model}=="nvme_card", ENV{ID_SERIAL_SHORT}="local-nvme-ssd-22"
+KERNEL=="nvme0n24*", ATTRS{model}=="nvme_card", ENV{ID_SERIAL_SHORT}="local-nvme-ssd-23"
```
Comment (hopkiw) [inline src/lib/udev/rules.d/65-gce-disk-naming.rules:47]: i will review this with you more in depth tomorrow, but the previous change was designed to remove explicit rules like these

Comment (yh117) [reply src/lib/udev/rules.d/65-gce-disk-naming.rules:47]: I tried the following as a one line solution, but it didn't work. Seems like udev doesn't have a good support on escaping double quotes https://github.com/systemd/systemd/issues/6835. This PR is made as a quick and easy fix.

```
KERNEL=="nvme*n*", ATTRS{model}=="nvme_card", PROGRAM="echo %k | sed \"s/nvme[0-9]\+n\([0-9]\+\).*/\1/g\"", ENV{ID_SERIAL_SHORT}="local-nvme-ssd-%c"
```

**Comment by yh117** [comment]
Friendly ping. Note that this is to fix an issue in prod. As-is, it will break when the disk has multiple partitions and do weird things.

**Comment by hopkiw** [comment]
this PR is pending until either you've got a properly dynamic fix or I have the time to write one. we won't merge this as-is, this codebase never receives short term hacks or fixes bc the code here is deployed widely and with no guaranteed update mechanisms. i'll try to get to writing one this week

**Comment by hopkiw** [comment]
OK, had some time to investigate. the rules work fine for whole disks because %n refers to the namespace ID, but the %n variable becomes the partition ID when dealing with partitions. so you get correct names for whole disks but striped output for partitions, because %n was used in the base name and in the partition name. there are a couple ways to fix this, it just means we can't blindly use %n to do our work for us. will update PR with a fix.

what a system with 4 NVME devices looks like:
```
google-local-nvme-ssd-0 -> ../../nvme0n1
google-local-nvme-ssd-1 -> ../../nvme0n2
google-local-nvme-ssd-2 -> ../../nvme0n3
google-local-nvme-ssd-3 -> ../../nvme0n4
```

adding 2 partitions to the first disk:
```
google-local-nvme-ssd-0 -> ../../nvme0n1
google-local-nvme-ssd-0-part1 -> ../../nvme0n1p1
google-local-nvme-ssd-1-part2 -> ../../nvme0n1p2
```

**Comment by hopkiw** [comment]
the fix here was only annoying in that we have to do it in two steps, because we want the id decremented from the namespace id for some reason. output with latest:
```
google-local-nvme-ssd-0 -> ../../nvme0n1
google-local-nvme-ssd-0-part1 -> ../../nvme0n1p1
google-local-nvme-ssd-0-part2 -> ../../nvme0n1p2
google-local-nvme-ssd-1 -> ../../nvme0n2
google-local-nvme-ssd-2 -> ../../nvme0n3
google-local-nvme-ssd-3 -> ../../nvme0n4
```

let's grab another reviewer since i'm also now part author of this change.

**Pull request merged**